### ISSUE TITLE
Wrapper Bindings Pre-Compiled Library

### ISSUE THREAD

**Issue by dOrgJelli** [Web3-API/prototype#700]
Pre-compile a portion of our wrapper bindings to a static library, which will later be linked in during a final build step within the compiler.

This will help make more of the bindings language-agnostic.

In order for this to be "worth while", the static library should contain **a significant** amount of code, that would otherwise be generated in the wrapper implementation language (Rust, TinyGo, AssemblyScript, etc).

**Comment by Niraj-Kamdar** [comment]
I agree with this, we can compile down almost all the code in wasm-as and dynamically link it.
